rs/nns/governance/src/network_economics_tests.rs

```diff
@@ -0,0 +1,38 @@
+use super::*;
+use pretty_assertions::assert_eq;
+
+#[test]
+fn test_inherit_from_recursively() {
+    let base = NetworkEconomics::with_default_values();
+
+    let changes = NetworkEconomics {
+        reject_cost_e8s: 99,
+
+        // This should not show up in the result.
+        neuron_management_fee_per_proposal_e8s: 0,
+
+        voting_power_economics: Some(VotingPowerEconomics {
+            start_reducing_voting_power_after_seconds: Some(42),
+
+            // This should not show up in the result.
+            clear_following_after_seconds: None,
+        }),
+
+        // This should not show up in the result.
+        neurons_fund_economics: None,
```

Comment:
I'd suggest adding a case where `Some(0)` is overriding, e.g., `Some(42)`

Reply:
Added Some(0). However, Some(0).inherit_from(Some(nonzero)) returns Some(nonzero). This is maybe surprising. If you really hate that, let's change it in a later PR. For now, I'm not going to try to address that, because this PR already solves the main objective (I.e. be able to change a deep subfield, one at a time).